servo

```diff
@@ -243,10 +247,9 @@ def run_driver(driver, app, req=None, describe=False, progress_cb: typing.Callab
                     continue # ignore blank lines (shouldn't be output, though)
                 try:
                     stdout = json.loads(stdout_line)
-                except Exception:
+                except Exception as e:
                     proc.terminate()
-                    # TODO: handle exception in json.loads?
-                    raise
+                    raise DriverOutputDecodingError("Failed decoding JSON stdout from '%' driver: %", driver, stdout_line) from e
```

Comment:
By the look of it, you expect string formatting to happens somewhere automatically.
I don't think that's the case, and you either need to define `DriverOutputDecodingError.__init__()` that is doing something meaningful with the args, or more probably just pass an already formatted string as a single argument:
```python
raise DriverOutputDecodingError(f"Failed decoding JSON stdout from '{driver}' driver: {stdout_line}") from e
```

Reply:
Good catch fixing

Reply:
Fixed. I will adopt `f` strings all over going forward